integrations/signup_test.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integration
```

Comment:
The motivation that led me to this way is that it depends on `/install` and some housekeeping works (creating tempdir, launching the program ... etc.). Imaging that we are testing 'merging a pull-request', We have to 

1. Set up a Gitea server from `/install`
0. Signup a user
0. Create a repo
0. Push a prepared repo to Gitea
0. Signup another user
0. Fork the repo on Gitea
0. Clone it
0. Make some changes to the cloned code, commit them and push them back
0. Create a pull request
0. Push the 'merge' button

That's what I had to do manually to test it reliably without depending on certain kept states.
Otherwise, I'd also go for the idiomatic unit test scheme.